prov/sockets/src/sock_fabric.c

```diff
@@ -319,7 +320,10 @@ static int sock_fabric(struct fi_fabric_attr *attr,
 	fab->fab_fid.ops = &sock_fab_ops;
 	*fabric = &fab->fab_fid;
 	atomic_init(&fab->ref, 0);
-	sock_fab_add_to_list(fab);
+#if ENABLE_DEBUG	
+	atomic_init(&fab->num_send_msg, 0);
+#endif
```

Comment:
As a general rule, it is preferable to avoid inserting ifdef statements into .c files.  It ends up producing code that is difficult to read and maintain.